### DISCUSSION TITLE
How to properly update a dependency on the first run of a week?

### DISCUSSION THREAD

**Discussion by DManstrator** [renovatebot/renovate#23204] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.28.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

About 3 months ago, we added a schedule to update Renovate itself.

```json
    {
      "matchPackageNames": [
        "renovate/renovate"
      ],
      "schedule": [
        "on the first day of the week"
      ]
    }
```

With that we wanted that the update comes the first time Renovate is running in a week but I guess this won't work since we got no updates in the last 3 months.

From the docs I found `before 4am on the first day of the month` which wouldn't actually work out (beside weekly vs. monthly) because Renovate doesn't run that early. I guess a time is needed for that and I would have to use the hour of when Renovate first runs, wouldn't I?

### What you tried so far.

Checked the docs and the logs.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: prAlreadyExisted=false (repository=config-repo, baseBranch=main, branch=renovate/renovate-renovate-35.x)
DEBUG: Checking schedule(on the first day of the week, Europe/Berlin) (repository=config-repo, baseBranch=main, branch=renovate/renovate-renovate-35.x)
DEBUG: Found timezone: Europe/Berlin (repository=config-repo, baseBranch=main, branch=renovate/renovate-renovate-35.x)
DEBUG: Adjusting now for timezone (repository=config-repo, baseBranch=main, branch=renovate/renovate-renovate-35.x)
DEBUG: Checking 1 schedule(s) (repository=config-repo, baseBranch=main, branch=renovate/renovate-renovate-35.x)
DEBUG: Checking schedule "on the first day of the week" (repository=config-repo, baseBranch=main, branch=renovate/renovate-renovate-35.x)
       "parsedSchedule": {"schedules": [{"d": [1]}], "exceptions": [], "error": -1}
DEBUG: Package not scheduled (repository=config-repo, baseBranch=main, branch=renovate/renovate-renovate-35.x)
DEBUG: Skipping branch creation as not within schedule (repository=config-repo, baseBranch=main, branch=renovate/renovate-renovate-35.x)
```
</details>

**Comment by rarkins** [comment, marked as answer]
You have to ensure that your bot is running during that time. If so and you still don't get updates then you need to capture the debug logs to work out why

**Reply by rarkins** [reply]
And use cron scheduling syntax instead if you're in doubt about the syntax/wording

**Reply by DManstrator** [reply]
So I guess the main problem was still that no time was used in the schedule. I changed it to `"schedule": [ "0 6 * * 1" ]` and created a pipeline schedule in GitLab with the exact same value (which will trigger Renovate to start). I hope that will work out.

**Reply by rarkins** [reply]
Don't specify minutes. Narrowest granularity allowed is an hour

**Reply by DManstrator** [reply]
Okay, fair. It could go wrong there. I changed it to `"schedule": [ "* 6 * * 1" ]` and if the GitLab schedule runs at 6 AM, it should work out. Thanks!
